app/src/main/java/io/supercharge/sccameraview/MainActivity.java

```diff
@@ -1,13 +1,103 @@
 package io.supercharge.sccameraview;
 
-import android.support.v7.app.AppCompatActivity;
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.FrameLayout;
+
+import com.example.sccameraview.BaseCameraView;
+import com.example.sccameraview.Camera1View;
+import com.example.sccameraview.Camera2View;
 
 public class MainActivity extends AppCompatActivity {
+    private static final int PERMISSION_REQUEST_CODE = 1;
+
+    private FrameLayout videoContainer;
+    private BaseCameraView cameraView;
+    private Button switchButton;
+    private Button recordButton;
+    private Button stopButton;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        if (isEveryPermissionGranted()) {
+            bindViews();
```

Comment:
You can bind the views without permission. The only thing is when you open the camera, everything else can be written outside of the `if`.